hbase-server/src/main/java/org/apache/hadoop/hbase/master/procedure/DeleteTableProcedure.java

```diff
@@ -100,6 +101,12 @@ protected Flow executeFromState(final MasterProcedureEnv env, DeleteTableState s
           // TODO: Move out... in the acquireLock()
           LOG.debug("Waiting for RIT for {}", this);
           regions = env.getAssignmentManager().getRegionStates().getRegionsOfTable(getTableName());
```

Comment:
I think here, we could add a method in RegionStates to get regions of table for disabling, so we do not need to interate on the region state nodes of this table again.